jwst/ramp_fitting/ramp_fit.py

```diff
@@ -303,92 +328,237 @@ def ols_ramp_fit(model, buffsize, save_opt, readnoise_model, gain_model,
-    opt_res.weights = (opt_res.inv_var_seg[:,:f_max_seg,:,:])**2.
-
-    var_p_s_4d[ var_p_s_4d < 0.] = 0.
-    var_r_s_4d[ var_r_s_4d < 0.] = 0.
-
-    opt_res.var_p_seg = var_p_s_4d
-    opt_res.var_r_seg = var_r_s_4d
+    opt_res.weights = (inv_var_both4[:,:f_max_seg,:,:])**2.
+    opt_res.var_p_seg = var_p4[:,:f_max_seg,:,:]
+    opt_res.var_r_seg = var_r4[:,:f_max_seg,:,:]
 
-    if save_opt: 
+    if save_opt:
         opt_model = opt_res.output_optional(model, effintim)
     else:
         opt_model = None
 
     # Divide slopes by total (summed over all integrations) effective
     #   integration time to give count rates.
-    c_rates = slopes / effintim
-
-    # Values in err_cube have been 1./variance, so take reciprocal
-    #    of non-zero values to write
-    err_cube[err_cube <= 0.] = MIN_ERR  # for pixels having no signal
-    err_cube = 1. / err_cube # has no zero values
+    c_rates = slope_dataset2 / effintim
 
     # Compress all integration's dq arrays to create 2D PIXELDDQ array for
     #   primary output
     final_pixeldq = dq_compress_final(dq_int, n_int)
 
     if n_int > 1:
-        int_model = utils.output_integ(model, slope_int, dq_int, effintim, var_p_int, var_r_int)
+        int_model = utils.output_integ(model, slope_int, dq_int, effintim,
+                                       var_p3, var_r3, var_both3)
     else:
         int_model = None
 
     tstop = time.time()
 
     log_stats(c_rates)
 
-    log.debug('Instrument: %s' % (instrume))
-    log.debug('Number of pixels in 2D array: %d' % (npix))
-    log.debug('Shape of 2D image: (%d, %d)' % (imshape))
-    log.debug('Shape of data cube: (%d, %d, %d)' % (orig_cubeshape))
-    log.debug('Buffer size (bytes): %d' % (buffsize))
-    log.debug('Number of rows per buffer: %d' % (nrows))
-    log.info('Number of groups per integration: %d' % (orig_nreads))
-    log.info('Number of integrations: %d' % (n_int))
-    log.debug('The execution time in seconds: %f' % (tstop - tstart))
+    log.debug('Instrument: %s', instrume)
+    log.debug('Number of pixels in 2D array: %d', npix)
+    log.debug('Shape of 2D image: (%d, %d)' %(imshape))
+    log.debug('Shape of data cube: (%d, %d, %d)' %(orig_cubeshape))
+    log.debug('Buffer size (bytes): %d', buffsize)
+    log.debug('Number of rows per buffer: %d', nrows)
+    log.info('Number of groups per integration: %d', orig_nreads)
+    log.info('Number of integrations: %d', n_int)
+    log.debug('The execution time in seconds: %f', tstop - tstart)
 
     # Create new model...
     new_model = datamodels.ImageModel(data=c_rates.astype(np.float32),
-            dq=final_pixeldq.astype(np.uint32), 
-            var_poisson=var_p_2d_all.astype(np.float32),
-            var_rnoise=var_r_2d_all.astype(np.float32), err=err_2d_all.copy())
+            dq=final_pixeldq.astype(np.uint32),
+            var_poisson=var_p2.astype(np.float32),
+            var_rnoise=var_r2.astype(np.float32), err=err_2d.copy())
```

Comment:
Is the copy really needed for err_2d, or would a simple view of the array work, as is done for the other arrays?

Reply:
No, copy() is not needed - fixed.